src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/AbstractFileOrTextStepExecution.java

```diff
@@ -1,21 +1,36 @@
 package org.jenkinsci.plugins.pipeline.utility.steps;
 
 import hudson.FilePath;
+import hudson.model.TaskListener;
 import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
 import org.jenkinsci.plugins.workflow.steps.MissingContextVariableException;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution;
 
+import javax.annotation.Nonnull;
 import javax.inject.Inject;
 
 import static org.apache.commons.lang.StringUtils.isNotBlank;
 
-public abstract class AbstractFileOrTextStepExecution<T> extends AbstractSynchronousNonBlockingStepExecution<T> {
-    @Inject
-    private transient AbstractFileOrTextStep fileOrTextStep;
+public abstract class AbstractFileOrTextStepExecution<T> extends SynchronousNonBlockingStepExecution<T> {
+
+    private AbstractFileOrTextStep fileOrTextStep;
+    protected transient TaskListener listener;
```

Comment:
OK I guess, though perhaps better to just have `doRun` impls look this up if and when they need it.